app/components/instruction/historical_events_component.rb

```diff
@@ -0,0 +1,106 @@
+class Instruction::HistoricalEventsComponent < ViewComponent::Base
+  attr_reader :authorization_request_event
+
+  def initialize(authorization_request_event:)
+    @authorization_request_event = authorization_request_event
+  end
+
+  def before_render
+    @text_content = case name
+                    when 'request_changes'
+                      helpers.simple_format(authorization_request_event.entity.reason)
+                    when 'applicant_message', 'instructor_message'
+                      helpers.simple_format(authorization_request_event.entity.body)
+                    when 'initial_submit_with_changes_on_prefilled_data', 'submit_with_changes'
+                      humanized_changelog
+                    when 'admin_update'
+                      humanized_changelog(from_admin: true)
+                    end
+  end
+
+  def name
+    case authorization_request_event.name
+    when 'submit'
+      changelog_presenter.event_name
+    when 'cancel_reopening'
+      name_for_cancel_reopening
+    else
+      authorization_request_event.name
+    end
+  end
+
+  def text
+    @text_content
+  end
+
+  # rubocop:disable Metrics/AbcSize
+  def transfer_text
+    return unless name == 'transfer'
+
+    if authorization_request_event.entity.from_type == 'User'
+      "#{authorization_request_event.entity.to.full_name} (#{authorization_request_event.entity.to.email})"
+    else
+      "l'organisation #{authorization_request_event.entity.to.raison_sociale} (numéro SIRET : #{authorization_request_event.entity.to.siret})"
+    end
+  end
+
+  # rubocop:enable Metrics/AbcSize
+
+  def message_content
+    @message_content ||= "#{message_summary} #{message_details}".html_safe
+  end
+
+  def message_summary
+    I18n.t(
+      "instruction.authorization_request_events.authorization_request_event.#{name}.message_summary",
+      user_full_name: authorization_request_event.user_full_name,
+      copied_from_authorization_request_id: copied_from_authorization_request_id,
+      transfer_text: transfer_text
+    ).html_safe
+  end
+
+  def message_details
+    return nil if text.blank?
+
+    I18n.t(
+      "instruction.authorization_request_events.authorization_request_event.#{name}.message_details",
+      text: text,
+      copied_from_authorization_request_id: copied_from_authorization_request_id
+    ).html_safe
+  end
+
+  def copied_from_authorization_request_id
+    return unless name == 'copy'
+
+    authorization_request_event.copied_from_request&.id
+  end
```

Comment:
```suggestion

  def copied_from_authorization_request_id
    authorization_request_event.copied_from_request_id
  end
```

par ailleurs (niptick, non-blocking) tu peux revert dans le modèle le lien avec le modèle qui imo est overkill